.github/workflows/pr-branch-build.yml

```diff
@@ -11,17 +11,21 @@ jobs:
 
     strategy:
       matrix:
-        python_version: ["3.7", "3.8", "3.9", "3.10"]
+        python_version: ["3.8", "3.9", "3.10", "3.11"]
 
     steps:
       - uses: actions/checkout@v3
 
+      - name: Install poetry
+        run: pipx install poetry
```

Comment:
Seems good to pin the version of Poetry that's installed. `1.5.1` is the current stable version.